src/Microsoft.AspNetCore.Mvc.Core/Internal/DefaultBindingMetadataProvider.cs

```diff
@@ -107,34 +107,34 @@ public void CreateBindingMetadata(BindingMetadataProviderContext context)
             }
         }
 
-        private class CompositePredicateProvider : IPropertyBindingPredicateProvider
+        private class CompositePropertyFilterProvider : IPropertyFilterProvider
```

Comment:
Why do we need two of these? Should be simple-enough that making one or other version `public` won't be painful later.